TPP.Core/Overlay/Events/RunInputEvents.cs

```diff
@@ -66,11 +67,81 @@ public sealed class AnarchyInputStop : IOverlayEvent
-        public ButtonPressUpdate(long numTotalButtonPresses) => NumTotalButtonPresses = numTotalButtonPresses;
+        public ButtonPressesCountUpdate(long numTotalButtonPresses) => NumTotalButtonPresses = numTotalButtonPresses;
+    }
+
+    [DataContract]
+    public readonly struct NewVote
+    {
+        [DataMember(Name = "command")] public string Command { get; init; }
+        [DataMember(Name = "user")] public User User { get; init; }
+        // [DataMember(Name = "button_sequence")] public InputSequence InputSequence { get; init; } // unused
+        // [DataMember(Name = "ts")] public Instant Timestamp { get; init; } // unused
+    }
+
+    [DataContract]
+    public readonly struct Vote
+    {
+        [DataMember(Name = "command")] public string Command { get; init; }
+        [DataMember(Name = "count")] public int Count { get; init; }
+        // [DataMember(Name = "button_sequence")] public InputSequence InputSequence { get; init; } // unused
+    }
+
+    [DataContract]
+    public sealed class DemocracyVotesUpdate : IOverlayEvent
+    {
+        public string OverlayEventType => "democracy_new_vote";
+
+        [DataMember(Name = "new_vote")] public NewVote NewVote { get; init; }
+        [DataMember(Name = "votes")] public List<Vote> Votes { get; init; }
+
+        public DemocracyVotesUpdate(
+            User newVoteUser,
+            InputSequence votedInput,
+            IReadOnlyDictionary<InputSequence, int> votes)
+        {
+            NewVote = new NewVote { User = newVoteUser, Command = votedInput.OriginalText };
+            Votes = votes
+                .Select(kvp => new Vote { Command = kvp.Key.OriginalText, Count = kvp.Value })
+                .OrderBy(vote => vote.Count)
+                .ToList();
+        }
+    }
+
+    [DataContract]
+    public sealed class DemocracyReset : IOverlayEvent
+    {
+        public string OverlayEventType => "democracy_reset";
+
+        [DataMember(Name = "vote_ends_at")] public Instant Timestamp { get; init; }
+        public DemocracyReset(Instant timestamp) => Timestamp = timestamp;
+    }
+
+    [DataContract]
+    public sealed class DemocracyVotingOver : IOverlayEvent
+    {
+        public string OverlayEventType => "democracy_voting_over";
+
+        [DataMember(Name = "winning_button_sequence")] public string WinningSequence { get; init; }
+        public DemocracyVotingOver(InputSequence input) => WinningSequence = input.OriginalText;
+    }
+
+    [DataContract]
+    public sealed class DemocracySequenceStart : IOverlayEvent
+    {
+        public string OverlayEventType => "democracy_sequence_start";
+
+        [DataMember(Name = "button_sequence")] public List<List<string>> Sequence { get; init; }
+        public DemocracySequenceStart(InputSequence input)
+        {
+            Sequence = input.InputSets
+                .Select(set => set.Inputs
+                    .Select(i => i.DisplayedText).ToList()).ToList();
```

Comment:
The old overlay relies on the input text to be regex-parseable to be recognized as a touch input and therefore displayed during execution, so this does not work for touchscreen aliases. Might be worth rethinking the wire format.
Note that in the old overlay the wire format for button sequences is shared across many places, and in particular with the `anarchy_input_start` event in regards to the touch detection logic (see `check_button_set_for_touchscreen` in old overlay `touchscreen.js`).